beamer/chain.py

```diff
@@ -303,15 +303,28 @@ def fill_request(request: Request, context: Context) -> None:
         log.info("Unable to fill request", balance=balance, request_amount=request.amount)
         return
 
+    allowance = context.config.token_checker.allowance(request.target_chain_id, token.address)
+    if allowance is None:
+        allowance = request.amount
+
+    if allowance < request.amount and allowance != -1:
+        log.info(
+            "Allowance for token smaller than requested amount",
+            token=token.address,
+            allowance=allowance,
+            request_amount=request.amount,
+        )
+        return
+
     if (
         token.functions.allowance(context.address, context.fill_manager.address).call()
         < request.amount
     ):
-        func = token.functions.approve(context.fill_manager.address, request.amount)
+        func = token.functions.approve(context.fill_manager.address, allowance)
         try:
             transact(func)
         except TransactionFailed as exc:
-            log.error("approve failed", request_id=request.id, exc=exc)
+            log.error("Approve failed", request_id=request.id, exc=exc)
```

Comment:
Nitpick: I prefer the previous spelling because it referred to the function name, `approve`, which I think is consistent with all the other logging calls in the agent.